src/cascadia/TerminalConnection/AzureConnection.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "AzureConnection.g.h"
+
+#include <cpprest/http_client.h>
+#include <cpprest/http_listener.h>
+#include <cpprest/ws_client.h>
+#include <mutex>
+#include <condition_variable>
+
+namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
+{
+    struct AzureConnection : AzureConnectionT<AzureConnection>
+    {
+        AzureConnection(const uint32_t rows, const uint32_t cols);
+
+        winrt::event_token TerminalOutput(TerminalConnection::TerminalOutputEventArgs const& handler);
+        void TerminalOutput(winrt::event_token const& token) noexcept;
+        winrt::event_token TerminalDisconnected(TerminalConnection::TerminalDisconnectedEventArgs const& handler);
+        void TerminalDisconnected(winrt::event_token const& token) noexcept;
+        void Start();
+        void WriteInput(hstring const& data);
+        void Resize(uint32_t rows, uint32_t columns);
+        void Close();
+
+    private:
+        winrt::event<TerminalConnection::TerminalOutputEventArgs> _outputHandlers;
+        winrt::event<TerminalConnection::TerminalDisconnectedEventArgs> _disconnectHandlers;
+
+        uint32_t _initialRows{};

```

Comment:
Generally, define all the default values for your variables here or define them all in the constructor. But don't define some of them and not the rest.

Reply:
And for the record, we've been having lots of discussions recently about how the "right" place to default initialize things is in the class definition

Reply:
_initialRows and _intialCols are defined in the constructor because the constructor takes in those as arguments. I will move the initializations of the URIs to the class definition though.